src/sql/src/plan/statement/ddl.rs

```diff
@@ -2992,27 +3001,129 @@ pub fn plan_create_connector(
         if_not_exists,
     } = stmt;
     let connector_inner = match connector {
-        CreateConnector::Kafka {
-            broker,
-            with_options,
-        } => {
-            let mut with_options = normalize::options(&with_options);
-            ConnectorInner::Kafka {
-                broker: broker.parse()?,
-                config_options: kafka_util::extract_config(&mut with_options)?,
-            }
-        }
+        CreateConnector::Kafka { broker, security } => ConnectorInner::Kafka {
+            broker: broker.parse()?,
+            // TODO resolve secrets from names to values
+            security: match security {
+                mz_sql_parser::ast::KafkaSecurityOptions::SSL {
+                    key,
+                    certificate,
+                    passphrase,
+                    authority,
+                } => {
+                    let key_id = if let Some(key) = key.as_ref() {
+                        let key_id = name_to_id(scx, key)?;
+                        depends_on.push(key_id);
+                        Some(key_id)
+                    } else {
+                        None
+                    };
+                    let certificate_id = if let Some(certificate) = certificate.as_ref() {
+                        let certificate_id = name_to_id(scx, certificate)?;
+                        depends_on.push(certificate_id);
+                        Some(certificate_id)
+                    } else {
+                        None
+                    };
+                    let passphrase_id = if let Some(passphrase) = passphrase.as_ref() {
+                        let passphrase_id = name_to_id(scx, passphrase)?;
+                        depends_on.push(passphrase_id);
+                        Some(passphrase_id)
+                    } else {
+                        None
+                    };
+                    let authority_id = if let Some(authority) = authority.as_ref() {
+                        let authority_id = name_to_id(scx, authority)?;
+                        depends_on.push(authority_id);
+                        Some(authority_id)
+                    } else {
+                        None
+                    };
+                    KafkaSecurityOptions::SSL {
+                        key: key_id.map_or_else(|| None, |i| Some(i.to_string())),
```

Comment:
I think these calls `to_string` should go in their respective `map` operations, and you should be able to just write this in such a way that the variables autocomplete the fields.